purchases/build.gradle

```diff
@@ -29,7 +29,7 @@ dependencies {
     testImplementation "io.mockk:mockk:$mockkVersion"
     testImplementation "org.assertj:assertj-core:$assertJVersion"
     testLatestDependenciesImplementation "com.android.billingclient:billing:$billingVersion"
-    testUnityIAPCompileOnly "com.android.billingclient:billing:$billingClient3Version"
+    testUnityIAPImplementation "com.android.billingclient:billing:$billingClient3Version"
```

Comment:
I had to change this because it was failing the tests https://circleci.com/gh/RevenueCat/purchases-android/7197?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link . The class couldn't be found. I think what's happening is that before, `testUnityIAPCompileOnly` was actually not necessary, since `unityIAPApi` was bringing the dependency to the tests. When changing it to `unityIAPCompileOnly`, the dependency is not added to the tests, but doing `testUnityIAPCompileOnly` doesn't work, because the `BillingFactory` looks for the `PurchasesUpdatedListener` during runtime and `CompileOnly` doesn't include the dependency in runtime so `PurchasesUpdatedListener` won't be available.

Reply:
so why do we only need implementation for tests? wouldn't we have the same problem running an app?

Reply:
great explanation, thank you